src/Legacy/Storage.php

```diff
@@ -401,10 +401,10 @@ public function saveContent(Content $content, $comment = null)
         }
 
         // Test to see if this is a new record, or an update
-        if (empty($fieldvalues['id'])) {
-            $create = true;
-        } else {
-            $create = false;
+        $create = empty($fieldvalues['id']);
+
+        if (!isset($fieldvalues['slug'])) {
+            $fieldvalues['slug'] = ''; // Prevent 'slug may not be NULL'
```

Comment:
> 1. create a slug (silently) in the background

Yes, I *think* this is the better way to go. You could just use the `getTitle()` helper to get the value of the first string field (or one called "title" if it exists), and Slugify that.